### ISSUE TITLE
Fedora actions failing to build.

### ISSUE THREAD

**Issue by ukbeast** [Frogging-Family/wine-tkg-git#957]
The last succeeded build was 3 months ago.
`/__w/wine-tkg-git/wine-tkg-git/wine-tkg-git/wine-tkg-scripts/prepare.sh: line 592: ../wine-staging-git/patches/patchinstall.sh: No such file or directory`

**Comment by apprehensions** [comment]
facing this as well
```
/home/sewn/.local/src/wine-tkg-git/wine-tkg-git/wine-tkg-scripts/prepare.sh: line 592: ../wine-staging-git/patches/patchinstall.sh: No such file or directory
fatal: Not a valid object name
 -> Using upstream-commit file
HEAD is now at df5f6f66de8 kernelbase: Re-implement EnumProcessModules on top of EnumProcessModulesEx.
 ==> WARNING: Detected non-GNU realpath (busybox?), please disable community patches in case of issues
 ==> WARNING: fshack (and associated patches) currently disabled until conflicts are resolved
 ==> WARNING: Hotfix: Fix for Riot Client updates
 ==> WARNING: Hotfix: Bcrypt fixes for RDR2
 ==> WARNING: Hotfix: Remove a TRACE in the DXGIGetDebugInterface1 stub causing segfaults.
 ==> WARNING: Hotfix: Fixup for Battle.net failing with a Qt error
 ==> WARNING: Hotfix: Fix for CL/opencl.h header recognition
 ==> WARNING: Hotfix: Fix NosTale mouse bug
 -> cherry picking...
 -> Hotfixing...
 -> Applying steam.patch
 -> Applying CSMT-toggle.patch
patch: can't open 'patches/patchinstall.sh': No such file or directory
 ==> ERROR: Patch application has failed. The error was logged to /home/sewn/.local/src/wine-tkg-git/wine-tkg-git/prepare.log for your convenience.
 -> Removed BIG_UGLY_FROGMINER - Ribbit
 -> Removed Proton-tkg token - Valve Ribbit
 -> exit cleanup done
```

**Comment by apprehensions** [comment]
this is not 'fedora actions', but is probably global

**Comment by ToddAWalter** [comment]
I'm experiencing this as well on Gentoo. 

`CSMT-toggle.patch -- ( Applied CSMT toggle logic patch )
can't find file to patch at input line 19
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|From f31cbcf4cb8fe1285091d8ab053febb115fc08af Mon Sep 17 00:00:00 2001
|From: Firerat <firer4t@googlemail.com>
|Date: Sat, 7 Apr 2018 01:05:29 +0100
|Subject: [PATCH] winecfg-Staging: Toggle upstream CSMT implementation
|
|  Logic on the toggle is now inverted
|    'on' disables upstream's CSMT
|   'off' lets upstream wine use its default
|---
| patches/patchinstall.sh                            |  2 +
| ...necfg-Toggle-upstream-CSMT-implementation.patch | 95 ++++++++++++++++++++++
| 2 files changed, 97 insertions(+)
| create mode 100644 patches/winecfg-Staging/0006-winecfg-Toggle-upstream-CSMT-implementation.patch
|
|diff --git a/patches/patchinstall.sh b/patches/patchinstall.sh
|index 0ba33ed3..9ffeffc4 100755
|--- a/patches/patchinstall.sh
|+++ b/patches/patchinstall.sh
`

**Comment by ToddAWalter** [comment]
I am experiencing this as well.  Excerpt from prepare.log (for which I can't seem to disable the markdown on):

CSMT-toggle.patch -- ( Applied CSMT toggle logic patch )
can't find file to patch at input line 19
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|From f31cbcf4cb8fe1285091d8ab053febb115fc08af Mon Sep 17 00:00:00 2001
|From: Firerat <firer4t@googlemail.com>
|Date: Sat, 7 Apr 2018 01:05:29 +0100
|Subject: [PATCH] winecfg-Staging: Toggle upstream CSMT implementation
|
|  Logic on the toggle is now inverted
|    'on' disables upstream's CSMT
|   'off' lets upstream wine use its default
|---
| patches/patchinstall.sh                            |  2 +
| ...necfg-Toggle-upstream-CSMT-implementation.patch | 95 ++++++++++++++++++++++
| 2 files changed, 97 insertions(+)
| create mode 100644 patches/winecfg-Staging/0006-winecfg-Toggle-upstream-CSMT-implementation.patch
|
|diff --git a/patches/patchinstall.sh b/patches/patchinstall.sh
|index 0ba33ed3..9ffeffc4 100755
|--- a/patches/patchinstall.sh
|+++ b/patches/patchinstall.sh

**Comment by Nanotwerp** [comment]
I'm having the same issue on Fedora Rawhide. The last time I tried to build (was successful) was two weeks ago.

**Comment by Nanotwerp** [comment]
My prepare.log:

```
fatal: destination path 'wine-git' already exists and is not an empty directory.
fatal: destination path 'wine-staging-git' already exists and is not an empty directory.
HEAD is now at 53e545fe Added odbc-remove-unixodbc patchset
HEAD is now at 464be65ceec Release 8.2.

a7efc770f7b1e1da959367e3186d871dce50c6e2 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

6f455e10afaf95f64008f45cdb5b794308275504 reverted (hotfix)
Auto-merging dlls/bcrypt/gnutls.c
Auto-merging dlls/bcrypt/tests/bcrypt.c

e1fb63168756c73ebd97a8c7f3b3aba00c52e444 reverted (hotfix)
Auto-merging dlls/bcrypt/gnutls.c

a88cc503653c06b612429d98e4e6174278687b94 reverted (hotfix)
Auto-merging dlls/bcrypt/tests/bcrypt.c

ccdef80543ff5c1f08a8aaff39c3f623522152df reverted (hotfix)

45b823114ce89ee56a36705a77f4f6bb9761dde4 reverted (hotfix)

96692a2c21583c50834d368d86e83511115474dd reverted (hotfix)
Auto-merging dlls/bcrypt/tests/bcrypt.c

5308911a82e5858f419354bcc6d9e35807134bc4 reverted (hotfix)

4c46bd922d286ed14442c548c53710613c5067d7 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

f95c2e4629f6cb15eabc19d71ccbe17c0758cd5d reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

ffad823e34f1f940530f72b7e17583da22d39d93 reverted (hotfix)

cb3cc2b59f47842fa8e46115354d8b4f446d2f5c reverted (hotfix)

11cc3ca15122c31fd5545372d749a6dafb0e7c60 reverted (hotfix)
Auto-merging dlls/bcrypt/tests/bcrypt.c

a4f8d28de77148f7e8a5ad0fe9376afaabb44d49 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

d75f651404028baa23d2591edf222837a167f790 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c
Auto-merging dlls/bcrypt/tests/bcrypt.c

d1291ebd4030c3ecc2f30199b963015077ab218c reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

ec85dfe3ab9ba56fbe8b3c30ef393d470e292a36 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

6f5028dd03eba2cfffc8a844e482040d4ca29655 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c
Auto-merging dlls/bcrypt/tests/bcrypt.c

6db0fc5ce8f556ca9a192a5f9167a3045338c9b0 reverted (hotfix)
Auto-merging dlls/rsaenh/implglue.c

ba1631ad885cd12034589fb60b1d90d8007ab3bb reverted (hotfix)

c7e0f923bc706b57dcb4a54c15733cc1acc12b07 reverted (hotfix)

7780caf4ee5747499643c511d612065756429cb7 reverted (hotfix)

430b9db051c0566424b0becee351591b6056894b reverted (hotfix)
Auto-merging dlls/appwiz.cpl/addons.c

c7b5778d83c688dd973efe503ad8646004e69296 reverted (hotfix)
Auto-merging dlls/appwiz.cpl/addons.c

4918be2c5905c849cc992e0968858365401c4a8e reverted (hotfix)
Auto-merging dlls/ncrypt/main.c

c6a75d01b52973ed0763ee1070634a2f2d111350 reverted (hotfix)
Auto-merging dlls/bcrypt/tests/bcrypt.c

fcb553ffc2caec728381fc577f1774432d3a969e reverted (hotfix)
Auto-merging include/bcrypt.h

5253c8d77e8e7dc5770981e97e67348f393b99b9 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c
Auto-merging dlls/bcrypt/tests/bcrypt.c
Auto-merging include/bcrypt.h

dd61c5638a0e597a488ca74e48f1ac97bc008f93 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_internal.h
Auto-merging dlls/bcrypt/bcrypt_main.c
Auto-merging dlls/bcrypt/gnutls.c
Auto-merging dlls/bcrypt/tests/bcrypt.c

48075d2a0874d368e617632bd5e875e4bc18c411 reverted (hotfix)
Auto-merging dlls/bcrypt/gnutls.c

1fa5bfd7dab270113c8fd2b48c838396719c639b reverted (hotfix)
Auto-merging dlls/bcrypt/gnutls.c
Auto-merging dlls/bcrypt/tests/bcrypt.c

5b860a44a0051cf779ec3391610a807e711d184f reverted (hotfix)
Auto-merging dlls/bcrypt/gnutls.c
Auto-merging dlls/bcrypt/tests/bcrypt.c

00dfa1bd04ae7f93ee7dd4235dd2439697efd318 reverted (hotfix)
Auto-merging dlls/bcrypt/gnutls.c
Auto-merging dlls/bcrypt/tests/bcrypt.c

777cbf06d2e8db298313982b7823acbc40dc05db reverted (hotfix)
Auto-merging dlls/bcrypt/gnutls.c

b352d353b41654ad2b94d46747c5d203acc417f3 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_internal.h
Auto-merging dlls/bcrypt/bcrypt_main.c
Auto-merging dlls/bcrypt/gnutls.c

0c2408464cfea0d3e5b073c56ac0d264d1abe576 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_internal.h
Auto-merging dlls/bcrypt/bcrypt_main.c
Auto-merging dlls/bcrypt/gnutls.c

c1ed9ca9b18267b2139fcf7fd68f24d2c6d7a2a3 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_internal.h
Auto-merging dlls/bcrypt/bcrypt_main.c
Auto-merging dlls/bcrypt/gnutls.c

9176251af425ba494c0334c5659aa47941ee7c85 reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

f3d4df60ff4afbf1983387a098ae96830384f53a reverted (hotfix)
Auto-merging dlls/bcrypt/bcrypt_main.c

CSMT-toggle.patch -- ( Applied CSMT toggle logic patch )
can't find file to patch at input line 19
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|From f31cbcf4cb8fe1285091d8ab053febb115fc08af Mon Sep 17 00:00:00 2001
|From: Firerat <firer4t@googlemail.com>
|Date: Sat, 7 Apr 2018 01:05:29 +0100
|Subject: [PATCH] winecfg-Staging: Toggle upstream CSMT implementation
|
|  Logic on the toggle is now inverted
|    'on' disables upstream's CSMT
|   'off' lets upstream wine use its default
|---
| patches/patchinstall.sh                            |  2 +
| ...necfg-Toggle-upstream-CSMT-implementation.patch | 95 ++++++++++++++++++++++
| 2 files changed, 97 insertions(+)
| create mode 100644 patches/winecfg-Staging/0006-winecfg-Toggle-upstream-CSMT-implementation.patch
|
|diff --git a/patches/patchinstall.sh b/patches/patchinstall.sh
|index 0ba33ed3..9ffeffc4 100755
|--- a/patches/patchinstall.sh
|+++ b/patches/patchinstall.sh
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
1 out of 1 hunk ignored
patching file patches/winecfg-Staging/0006-winecfg-Toggle-upstream-CSMT-implementation.patch
```

**Comment by papaj-na-wrotkach** [comment]
I have the same issue on Manjaro Linux (pc) and Arch Linux (GH Actions container)

**Comment by KokoseiJ** [comment]
This is probably because wine-staging has removed patchinstall.sh in [this commit.](https://github.com/wine-staging/wine-staging/commit/c1b4af92f74d7bd330003d39d0bb1f966fdb70a9) I tried to get around this by changing every usage of patches/patchinstall.sh to staging/patchinstall.py, and removing patchinstall.sh section in CSMT-toggle.patch, but I still encountered a problem applying LAA-unix-staging.patch:
```
LAA-unix-staging.patch -- ( Applied large address aware override support (legacy) )
patching file dlls/kernel32/heap.c
Hunk #1 succeeded at 65 with fuzz 2 (offset 21 lines).
Hunk #2 succeeded at 429 (offset -117 lines).
Hunk #3 succeeded at 471 (offset -119 lines).
patching file dlls/ntdll/ntdll.spec
Hunk #1 succeeded at 1708 (offset 83 lines).
patching file dlls/ntdll/unix/server.c
Hunk #1 succeeded at 1672 (offset 201 lines).
patching file dlls/ntdll/unix/unix_private.h
Hunk #1 succeeded at 503 with fuzz 2 (offset 33 lines).
patching file dlls/ntdll/unix/virtual.c
Hunk #1 succeeded at 3441 with fuzz 1 (offset 21 lines).
patching file dlls/ntdll/unix/loader.c
Hunk #1 FAILED at 347.
1 out of 1 hunk FAILED -- saving rejects to file dlls/ntdll/unix/loader.c.rej
```

**Comment by mickski56** [comment]
Hi
Modify LAA-unix-staging.patch to contain this
```
--- a/dlls/ntdll/unix/loader.c	2023-02-19 09:38:13.368434169 +0000
+++ b/dlls/ntdll/unix/loader.c	2023-02-19 09:39:05.475877423 +0000
@@ -358,6 +358,7 @@
     NtWriteFileGather,
     NtWriteVirtualMemory,
     NtYieldExecution,
+    __wine_needs_override_large_address_aware,
     wine_nt_to_unix_file_name,
     wine_unix_to_nt_file_name,
 };
```
instead of this 
```
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -346,6 +346,7 @@
     NtWriteVirtualMemory,
     NtYieldExecution,
     __wine_dbg_write,
+    __wine_needs_override_large_address_aware,
     __wine_unix_call,
     __wine_unix_spawnvp,
     wine_nt_to_unix_file_name,
```

**Comment by IroAlexis** [comment]
Normally, this issue should be fixed/closed by 8beb332a9e653f1c91a267fdbd33195cb0347225. You just have to wait for the IC to be triggered again.
